BFS/jordan/Solution.java

```diff
@@ -0,0 +1,74 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Solution {
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		GraphNode start = new GraphNode(0);
+		GraphNode mid1 = new GraphNode(1);
+		GraphNode mid2 = new GraphNode(2);
+		GraphNode mid3 = new GraphNode(3);
+		GraphNode finish = new GraphNode(4);
+		
+		start.connectedNodes.add(mid1);
+		mid1.connectedNodes.add(mid2);
+		mid2.connectedNodes.add(mid3);
+		mid3.connectedNodes.add(finish);
+		
+		ArrayList<GraphNode> path = shortestPath(start, finish);
+		
+		for(GraphNode node : path) {
+			System.out.print(node.val);
+			
+			if(node != finish)
+				System.out.print(" -> ");
+		}
+		
+		System.out.println();
+	}
+
+	//Shortest Path between nodes using BFS
+	private static ArrayList<GraphNode> shortestPath(GraphNode start, GraphNode finish){
+		if(start == null || finish == null)
+			return null;
+		
+		HashMap<GraphNode,GraphNode> map = new HashMap<GraphNode, GraphNode>();
+		ArrayList<GraphNode> queue = new ArrayList<GraphNode>();
+		map.put(start, null); 
+		queue.add(start);
+		
+		if(start == finish)
+			return queue;
+		
+		while(!queue.isEmpty()) {
```

Comment:
Just a personal thing, I am trying to write code such that is a bit more readable and I can't tell if 
while(queue.isEmpty() == False)
is more readable than 
while(!queue.isEmpty())